common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -103,12 +113,63 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
         iframeElement.remove()
       }
     }
+
+    def cleanEmbedElement(document: Document) = {
+      for {
+        audioElement <- document.getElementsByClass("element-embed")
+        iframeElement <- audioElement.getElementsByTag("iframe")
+        trackId <- getTrackIdFromUrl(iframeElement.attr("src"))
+      } yield {
+        val soundcloudElement = createElement(document, trackId)
+        audioElement.appendChild(soundcloudElement)
+        iframeElement.remove()
+      }
+    }
```

Comment:
Bonus points if at the same time you can avoid the side effects in the for comprehension (although this was already there so wouldn't worry too much).